### ISSUE TITLE
inconsitencies of colour with multiple windows

### ISSUE THREAD

**Issue by grahamperrin** [mozilla/testpilot-containers#772]
## Steps

1. have just one window, with a contained tab
2. open a new window
3. use the containers button to edit the colour of that container

## Expected

- the colour of the icon, and the name of the container, in the first window should change to match the more colour in the second

## Actual result

- one container with two colours.

**Comment by jonathanKingston** [comment]
I also don't see the underline update. Are you using Nightly? What OS are you using?

**Comment by grahamperrin** [comment]
```text
[grahamperrin@momh167-gjp4-hpelitebook8570p-freebsd] ~% date ; uname -v
Tue 29 Aug 2017 00:05:41 BST
FreeBSD 12.0-CURRENT #0 r320869: Mon Jul 10 13:57:55 UTC 2017     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC 
[grahamperrin@momh167-gjp4-hpelitebook8570p-freebsd] ~% pkg info firefox | grep -i version
Version        : 55.0.3_1,1
[grahamperrin@momh167-gjp4-hpelitebook8570p-freebsd] ~% 
```

If I disable Vertical Tabs Reloaded then the colour of a container appears  as _over_lines, above its contained tabs. Whilst the colour of elements within the location bar is bugged, overlines are not bugged.

**Comment by grahamperrin** [comment]
Name of container changed: 

- from _Hinge_
- to _Bracket_

Voilà: 

![2017-08-29 00 24 40 hinge and bracket](https://user-images.githubusercontent.com/192271/29797705-a8284826-8c50-11e7-8178-5d0198c809fd.png)

Hinge & Bracket `:-)`

**Comment by jonathanKingston** [comment]
@grahamperrin that looks like a Sea containers issue, we just added the API hooks to observe container changes.
The icon is missing in 55 as they aren't present in central until 57.
I think what I just reproduced is a different issue than what you have.

**Comment by grahamperrin** [comment]
> …  icon is missing in 55 as they aren't present in central until 57. …

Thanks for the reminder! 

I threw in the Sea Containers aspect for good measure. Ignoring that (to the left): 

- look to the centre of the screenshot, the arrow pointer
- then up from the purple of the _Bracket_ container (in the dropdown from the button of _Testpilot containers_ 3.1.0) 
- to the green of the word _Hinge_ in the location bar 

– it's the same container. Hinge ≠ Bracket and green ≠ pink.

**Comment by jonathanKingston** [comment]
Yeah ok that looks more like what was working before and what I am seeing. My hunch is we fixed that before in the extension only and then removed it expecting it to work in central.

This will require a platform fix.

**Comment by grahamperrin** [comment]
Also bugged: single-window environments. 

For example, following a change to red: 

![2017-09-01 03 49 51](https://user-images.githubusercontent.com/192271/29953615-b3edf5ae-8ec9-11e7-8691-7a4cec9ff1bd.png)

![2017-09-01 03 50 55](https://user-images.githubusercontent.com/192271/29953614-b3e97790-8ec9-11e7-91a0-a61d63b3d98d.png)
